sensor_node1:
 ros__parameters:
  # IP address for IP connect. "" is default.
  IP_Address: ""

  # Device ID support for multiple sensors. "" is default. "1" or "2" can be used for multiple sensors
  ServiceID: "1"

  # General FPS, Default: -1 (can be overriden for specific channels)
  FPS: 15

  # General default resolution. 0 - Default (Full), 1 - Binning, 2 - VerticalBInning, 3 - Full
  Resolution: 1

  # -1 - Default
  RGBChannel: -1

  # 0 - Default (Full), 1 - Binning, 2 - VerticalBInning, 3 - Full
  RGBResolution: 1

  #eDefault - 0        Provides CImageFrames with CImageFormat::eBGRA, on Android CImageFormat::eRGBA
  #eRaw     - 1        Provides CImageFrames as streamed by Inuitive chip, no additional processing on Host
  #eBGRA    - 2        Provides CImageFrames of CImageFormat::eBGRA format
  #eRGBA    - 3        Provides CImageFrames of CImageFormat::eRGBA format
  #eBGR     - 4        Provides CImageFrame of CImageFormat::eBGR format
  RGBOutputFormat: 4

  # 0 - none, 1 - eRegistered, 4 - eGammaCorrect  (RGB always undistorted, you can change it in InuServiceParams.xml)
  RGBPostProcessing: 0

  # Depth channel to register, Default: -1 (verify that output format is registered)
  RGBRegistrationChannel: -1

  # Trackiing camera interleave mode: -1 - default interleaved, 1 - right camera, 2 - left camera
  TrackingCameraInterleaveMode: 1

  # 0 - Default (Full), 1 - Binning, 2 - VerticalBInning, 3 - Full
  DepthResolution: 1

  # Default  (3 or 5)
  DepthChannel: 3

  # 0 - depth, 1 - Raw (disparity)
  DepthOutputFormat: 0

  # 0 - none, 2 - Blob, 4 - Temporal, 8 - Outlier Removal, 16 - Hole Fill, 32 - Static Temporal, 255 - default as defined by the host
  DepthPostProcessing: 0

  # Enable depth registration, Default: 0 (to activate depth registration set to 1)
  DepthRegistarionActive: 0

  # RGB channel to register, Default: -1 (-1: do not register; 4: register to RGB channel 4, 9: register to RGB channel 9)
  DepthRegistarionChannel: -1

  #Show Depth with Color on/off flag
  Show_Color: false

  # Confidence [0ï¼Œ15], can dynamic change through API
  Confidence: 0

  # 0 - Default(XYZ) ; 1 - XYZRGBa (with registration)
  PointCloudFormat: 0

  # RGB channel to register, 0 - IR, 4 - RGB
  PointCloudRegistrationChannel: 4

  # When point cloud is generated by the FW, point cloud format is in short so it is necessary to convert to float to be able to present in rviz
  # Note: not all sensor models support point cloud from FW. In the case of M4.5 point cloud is generated in the FW
  ConvertPointCloudShortToFloat: true

  # SSD or YOLOV3
  ObjectDetectionType: "SSD"

  # Auto exposure on/off flag
  AutoExposure: true

  # SensorID for exposure control
  SensorID: 0

  # Sensor Exposure time control
  ExposureControl: 3500

  # Sensor Gain control
  Gain: 25

  # ROI on/off flag
  UseROI: false

  # Auto Exposure ROI top left X position
  ROITopLeftX: 0

  # Auto Exposure ROI top left X position
  ROITopLeftY: 0

  # Auto Exposure ROI bottom right X position
  ROIBottomRightX: 0

  # Auto Exposure ROI bottom right Y position
  ROIBottomRightY: 0

  # RGB channel ID (Typically 0, 4 or 9), -1 default
  PCLRGBChannelID: -1

  # 0 - None, 1 - Registered
  PCLOutputFormat: 0

  # SLAM path size, Endless path is the default: -1
  SLAMPathSize: -1

  # Whether simplify two times
  Simplify_2: false

  # Temporal Filter on/off flag
  Temporal_Filter_ROS: false

  # Temporal Filter on/off flag
  Temporal_Filter_SDK: false

  # Passthrough Filter on/off flag
  Passthrough_Filter_ROS: false

  # Passthrough Filter on/off flag
  Passthrough_Filter_SDK: false

  # Show Outlier Removal on/off flag
  Outlier_Remove_Fitler_ROS: false

  # Show Outlier Removal on/off flag
  Outlier_Remove_Fitler_SDK: false

  # Hole fill Filter on/off flag
  Hole_Fill_Filter_ROS: false

  # Hole fill Filter on/off flag
  Hole_Fill_Filter_SDK: false

  # Temporal2 Filter  on/off flag
  Temporal2_Filter_ROS: false

  # Temporal2 Filter alpha
  Temporal2_Filter_alpha: 40

  # Teporal2 Filter delta
  Temporal2_Filter_delta: 20

  # Edge preserve on/off flag
  Edge_Preserve_ROS: false

  # Edge preserve alpha
  Edge_Preserve_alpha: 50

  # Edge preserve delta
  Edge_Preserve_delta: 20

  # Temporal_BufferFrames
  Temporal_BufferFrames: 6

  # Process All Pixel on/off flag
  Temporal_ProcessAllPixel: false

  # Threads to process
  Temporal_Threads: 1


  # Passthrough Filter ZMin
  PT_zmin: 40

  # Passthrough Filter ZMax
  PT_zmax: 6000

  # Passthrough Filter ZMin
  PT_xmin: -1

  # Passthrough Filter ZMax
  PT_xmax: -1

  # Passthrough Filter ZMin
  PT_ymin: -1

  # Passthrough Filter ZMax
  PT_ymax: -1

  # Threshold of max component to remove
  OR_percent: 10

  # Min dist(z,mm) between components
  OR_min_dist: 35

  #Max number of pixels to remove
  OR_max_remove: 50

  # HF max radius
  HF_maxradius: 25

  # 3D coordinates divided by this value
  PC_scale: 1000

  # Whether output ordered point cloud
  PC_ordered: false

  # Enable depth 2 point cloud stream
  Dept2PcSupportted: true

